tests/test_ipynb.py

```diff
@@ -0,0 +1,286 @@
+from black import NothingChanged, format_cell, format_ipynb_string
+import os
+from tests.util import DEFAULT_MODE
+import pytest
+import subprocess
+
+pytestmark = pytest.mark.jupyter
+
+
+def test_noop() -> None:
+    src = 'foo = "a"'
+    with pytest.raises(NothingChanged):
+        format_cell(src, mode=DEFAULT_MODE)
+
+
+def test_trailing_semicolon() -> None:
+    src = 'foo = "a" ;'
+    result = format_cell(src, mode=DEFAULT_MODE)
+    expected = 'foo = "a";'
+    assert result == expected
+
+
+def test_trailing_semicolon_with_comment() -> None:
+    src = 'foo = "a" ;  # bar'
+    result = format_cell(src, mode=DEFAULT_MODE)
+    expected = 'foo = "a";  # bar'
+    assert result == expected
+
+
+def test_trailing_semicolon_noop() -> None:
+    src = 'foo = "a";'
+    with pytest.raises(NothingChanged):
+        format_cell(src, mode=DEFAULT_MODE)
+
+
+def test_cell_magic() -> None:
+    src = "%%time\nfoo =bar"
+    result = format_cell(src, mode=DEFAULT_MODE)
+    expected = "%%time\nfoo = bar"
+    assert result == expected
+
+
+def test_cell_magic_noop() -> None:
+    src = "%%time\n2 + 2"
+    with pytest.raises(NothingChanged):
+        format_cell(src, mode=DEFAULT_MODE)
+
+
+@pytest.mark.parametrize(
+    "src, expected",
+    (
+        pytest.param("ls =!ls", "ls = !ls", id="System assignment"),
```

Comment:
What if the input already contains `get_ipython().system` or similar? The output should still be `get_ipython().system`, not something transformed back into magics.

Reply:
That's an excellent point, thanks - looks like they can't be distinguished after `TransformerManager` has transformed the cell:
```
>>> TransformerManager().transform_cell("get_ipython().system('ls')")
"get_ipython().system('ls')\n"
>>> TransformerManager().transform_cell("!ls")
"get_ipython().system('ls')\n"
```

The only solution I can think of is ignore cells which contain these transformed magics